recipe/meta.yaml

```
@@ -0,0 +1,51 @@
+{% set name = "pinentry" %}
+{% set version = "1.3.1" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/gpg/{{ name }}/archive/refs/tags/{{ name }}-{{ version }}.tar.gz
+  sha256: 3cd7e17a3c0008526c28f1ab6a05f865a7ba7d8c5e4af73d7103fbbb25349833
+
+build:
+  number: 0
+  missing_dso_whitelist:
+    - $RPATH/libassuan.9.dylib  # [osx]
+    - $RPATH/libgpg-error.0.dylib  # [osx]
+    - $RPATH/libassuan.so.9  # [linux]
+    - $RPATH/libgpg-error.so.0  # [linux]
+  # libgpg-error does not exist on the s390x and win platforms
+  skip: True  # [win]
+  skip: True  # [s390x]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - make
+    - libtool
+    - automake
+    - gettext
+    - libassuan >=3.0.1,<4
```

Shared libraries in the `host` section: https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#host

```suggestion
    host:
    - libassuan 3.0.1
```

---

You need an exact pinning, here, `3.0.1` (the only one available) -- as the linter is warning you.

---

In the host section you are essentially declaring the lowest API of the library you want your code to use.  The run section can then allow a much wider range because we're confident the library in question (`libassuan`, here) is highly backwards compatible.  Sometimes we're not so confident and we set (increasingly) tight(er) upper bounds.